java-checks/src/main/java/org/sonar/java/checks/WeakSSLContextCheck.java

```diff
@@ -86,7 +103,18 @@ public void visitNode(Tree tree) {
           .skip(1)
           .map(secondary -> new JavaFileScannerContext.Location("Other weak protocol.", secondary))
           .toList();
-        reportIssue(unsecureVersions.get(0), "Change this code to use a stronger protocol.", secondaries, null);
+        reportIssue(unsecureVersions.get(0), ISSUE_MESSAGE, secondaries, null);
+      }
+    } else if (OPTIONS_ENABLED_PROTOCOLS.matches(mit)) {
+      ExpressionTree argument = arguments.get(0);
+      if (argument instanceof MethodInvocationTree methodInvocation && SET_OF.matches(methodInvocation)) {
```

Comment:
I would argue we can perhaps skip the second match using `SET_OF` and just look in the AST node for string literals without this check. There can be many different ways of creating a new set (including custom methods) and I don't think we risk a lot of FPs by checking for the weak TLS strings in the children of the first argument, no matter how exactly the first argument creates the set. WDYT?